pkg/layer/group_builder_test.go

```diff
@@ -0,0 +1,15 @@
+package layer
+
+import (
+	"testing"
+
+	"github.com/reearth/reearth-backend/pkg/id"
+	"github.com/reearth/reearth-backend/pkg/tag"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGroupBuilder_Tags(t *testing.T) {
+	tags := []id.TagID{id.NewTagID()}
+	b := NewGroup().NewID().Tags(tag.NewListFromTags(tags)).MustBuild()
+	assert.Equal(t, tags, b.Tags().Tags())
```

Comment:
```suggestion
	b := NewGroup().NewID().Tags(tag.NewListFromTags(tags))
	assert.Same(t, tags, b.l.tags)
```

Let's think a little more carefully about whether we should use Equal or Same for assertions. If the implementation is just an assignment, then Same should be used to assert that the value being assigned is exactly the same as the expected value. On the other hand, if there is a possibility that the value will be copied, then Equal is fine. If you use Equal for everything, unit tests will not be able to detect if a value is unintentionally copied when testing an implementation of a function that does not copy values.

Also, accessing private fields of TagGroupBuilder and TagGroup is ok because they are just the unit test target.

Reply:
@rot1024 I believe equal is good here because tags are passed by value (please see `NewListFromTags(tags []id.TagID)`) but anyway I changed the test unit logic

Reply:
Yes, right. `Same` doesn't work for values (not references). But we have two choices to  assert a reference: Same or Equal.